internal/syncd/integrations/github/listener.go

```diff
@@ -0,0 +1,77 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+
+	"ship-it/internal/syncd"
+	"ship-it/internal/syncd/middleware"
+
+	"github.com/Wattpad/sqsconsumer"
+	sqsmiddleware "github.com/Wattpad/sqsconsumer/middleware"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/metrics"
+	"github.com/pkg/errors"
+	"k8s.io/helm/pkg/chartutil"
+)
+
+type githubDownloader interface {
+	BufferDirectory(ctx context.Context, repo, path, ref string) ([]*chartutil.BufferedFile, error)
+}
+
+type ChartListener struct {
+	downloader githubDownloader
+	logger     log.Logger
+	service    *sqsconsumer.SQSService
+	timer      metrics.Histogram
+}
+
+func NewListener(l log.Logger, h metrics.Histogram, org string, r RepositoriesService, queue string, sqs sqsconsumer.SQSAPI) (*ChartListener, error) {
+	svc, err := sqsconsumer.NewSQSService(queue, sqs)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ChartListener{
+		downloader: newDownloader(r, org),
+		logger:     log.With(l, "worker", "github"),
+		service:    svc,
+		timer:      h.With("worker", "github", "queue", queue),
```

Comment:
The `queue` tag is kinda pointless, the `github` worker only uses the single queue so it's redundant. Does it make sense to keep it or should I remove?

Reply:
As we discussed, it is easy to add if needed and can be removed for now.